### PR TITLE
Add script to automate builds

### PR DESCRIPTION
**Pull request by loremattei**
This PR adds a lane to automate the pre-build checks and the build itself.
This PR builds on top of https://github.com/Automattic/simplenote-android/pull/749 and should be merged after that one. 

# To Test
1. Run `bundle install`.
2. Run `bundle exec fastlane build_release`.
3. When prompted, verify that the version is correct and confirm.
4. If prompted, update the secret file.
5. Verify that the script finishes without errors.
6. Verify that the apk has been generated in the `build` folder.
7. Test the apk.

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -8,16 +8,277 @@ platform :android do
...
+  #####################################################################################
+  # build_bundle
+  # -----------------------------------------------------------------------------------
+  # This lane builds an app bundle
+  # -----------------------------------------------------------------------------------
+  # Usage:
+  # bundle exec fastlane build_bundle [skip_confirm:<skip confirm>]
+  #####################################################################################
+  desc "Builds an app bundle"
+  lane :build_apk do | options |
```
Comment (jtreanor) [inline fastlane/Fastfile]: Is there a reason we are building an apk rather than an app bundle? I guess we just haven't gotten around to that yet.

Comment (theck13) [reply fastlane/Fastfile]: Yep, we are still releasing the full APK.  There isn't a large size savings when using app bundle for Simplenote Android.  The app is only 3.8MB at the moment and we would reduce the download size by about 12%.

Comment (jtreanor) [reply fastlane/Fastfile]: Makes sense. We may want to make the change at some point anyway for consistency but not a big deal.

Comment (loremattei) [reply fastlane/Fastfile]: Oh yes, just something to do.

**Pull request merged**